src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Test.Common;
+using System.Runtime.InteropServices;
+
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class SocketOptionNameTest
+    {
+        private static bool IsWindows10Platform
+        {
+            get
+            {
+                return Capability.IsWindows10Platform();
+            }
+        }
+
+        private static bool IsNotWindows10Platform
+        {
+            get
+            {
+                return !Capability.IsWindows10Platform();
+            }
+        }
+
+        [ConditionalFact("IsNotWindows10Platform")]
+        public void ReuseUnicastPort_CreateSocketGetOption_NotWindows10Platform_Throws()
+        {
+            var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
We can get 2 different exceptions thrown if we're not Windows10, i.e. the "Capability.ReuseUnicastPortSupport' is not present.

On non-Windows10, it throws SocketException when get or set the socket option.

On *NIX systems, it throws PlatformNotSupportedException.

So, we still need to check for OS version in the tests in order to test against the exception that is being thrown.